File: fastFramework/src/main/java/com/tesla/framework/ui/fragment/itemview/IItemViewCreator.java
package com.tesla.framework.ui.fragment.itemview;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import java.io.Serializable;

/**
 * Created by jerryliu on 2017/4/3.
 */

public interface IItemViewCreator<T extends Serializable> {

    View newContentView(LayoutInflater inflater, ViewGroup parent, int viewType);

    IITemView<T> newItemView(View contentView, int viewType);

}

File: fastFramework/src/main/java/com/tesla/framework/ui/fragment/recycleview/IItemViewCreator.kt
package com.tesla.framework.ui.fragment.recycleview

import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView

/**
 * Created by Jerry on 2023/11/26.
 */
interface IItemViewCreator<VH:RecyclerView.ViewHolder> {
    fun onCreateViewHolder(parent: ViewGroup, viewType: Int): VH

    fun onBindViewHolder(holder: VH, position: Int)

}